src/lib/components/Swap/SwapPropValidator.tsx

```diff
@@ -15,7 +16,7 @@ function isAddressOrAddressMap(addressOrMap: DefaultAddress): boolean {
   return false
 }
 
-type ValidatorProps = PropsWithChildren<SwapProps>
+type ValidatorProps = PropsWithChildren<DefaultTokenAmounts & FeeOptions>
```

Comment:
Is this change necessary? It looks like it does almost the same thing:

`export interface SwapProps extends DefaultTokenAmounts, FeeOptions {`

Reply:
This prevents circular dependencies (SwapPropValidator imports Swap, and Swap imports SwapPropValidator). I think typescript allows it with types, but it still leads to weird coupling.